Feat/theme button styles

danielcj2 opened the pull request:
- [x] Refactored button components
- [x] Defined button variants in the theme file  
- [x] Fixed minor theme issues

ajhollid reviewed (changes requested):
Looks good to me, but the create monitor button's color isn't correct anymore:

![image](https://github.com/user-attachments/assets/a26f4693-71a1-4498-8ede-1eb5fc3446d3)

That's the only one I've found

danielcj2 commented:
> Looks good to me, but the create monitor button's color isn't correct anymore:
> 
> ![image](https://private-user-images.githubusercontent.com/8007637/363446513-a26f4693-71a1-4498-8ede-1eb5fc3446d3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUxNDg5MzQsIm5iZiI6MTcyNTE0ODYzNCwicGF0aCI6Ii84MDA3NjM3LzM2MzQ0NjUxMy1hMjZmNDY5My03MWExLTQ0OTgtOGVkZS0xZWI1ZmMzNDQ2ZDMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgzMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MzFUMjM1NzE0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDJiYjM5MWE5MTkxMDIyY2FjZGExY2FkZDY2MDk3OWQ0MmUxMTA4MGEwYzg3ZWMxNGYzODcxMGQ2OTc0YzQwZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.issa1iZsXOH6G-nLAD05v3vPs4ARX5VDA8N62IXlPJI)
> 
> That's the only one I've found

Woops, added the wrong variant prop. Fixed now.

ajhollid reviewed (approved):
Thanks for making changes!
